torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -427,7 +427,27 @@ void ProcessGroupAgent::enqueueRecv(RecvWork work) {
           auto futureResponse = cb_->operator()(message);
           if (futureResponse->completed()) {
             if (!futureResponse->hasError()) {
-              send(work.from_, std::move(*futureResponse).moveValue());
+              // Hack: Check for device type here since throwing an exception in
+              // the send() below is not handled by enqueueRecv yet. Should be
+              // fixed as part of
+              // https://github.com/pytorch/pytorch/issues/25516.
+              const auto& respMessage = futureResponse->wait();
+              std::string errorMsg;
+              for (const auto& tensor : respMessage.tensors()) {
+                if (!tensor.device().is_cpu()) {
+                  errorMsg = c10::str(
+                      "ProcessGroup RPC backend only supports",
```

Comment:
would it be worth adding tensor.device().str() to the errorMsg, so we know which non-cpu device the tensor is from?